auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackIDPPrompt.java

```diff
@@ -186,7 +194,32 @@ public void onComplete(@NonNull Task<AuthResult> task) {
             authResultTask
                     .addOnFailureListener(
                             new TaskFailureLogger(TAG, "Error linking with credential"))
-                    .addOnCompleteListener(new FinishListener());
+                    .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            FirebaseAuth auth = FirebaseAuth.getInstance();
+                            if (!task.isSuccessful() && task.getException() instanceof FirebaseAuthUserCollisionException) {
```

Comment:
nit: prefer separate `onSuccess` and `onFailure` listeners here, rather than checking `isSuccessful()` in the `onCompleteListener`